src/lib/device-types/storage/gh.ts

```diff
@@ -0,0 +1,42 @@
+import * as Octokit from '@octokit/rest';
+import * as _ from 'lodash';
+import * as Promise from 'bluebird';
+import { captureException } from '../../../platform/errors';
+
+import {
+	DEVICE_TYPE_REPO_NAME,
+	DEVICE_TYPE_REPO_OWNER,
+	DEVICE_TYPE_REPO_PATH,
+	DEVICE_TYPE_REPO_REF,
+} from '../../config';
+
+interface FetchDeviceTypeOptions {
+	owner: string;
+	repo: string;
+	path: string;
+	ref?: string;
+}
+
+export const fetchExternalDeviceTypes = (
+	options?: FetchDeviceTypeOptions,
+): Promise<string[]> => {
+	const realOptions = options || {
```

Comment:
This could be a default value in the argument, except if you were meant to do `_.defaults(options || {}, {...})` :thinking: